bin/check_samplesheet.py

```diff
@@ -83,46 +89,56 @@ def _validate_sample(self, row):
             raise AssertionError(
-                f"The FASTQ file has an unrecognized extension: {filename}\n"
-                f"It should be one of: {', '.join(self.VALID_FORMATS)}"
+                f"The datatype is unrecognized: {row[self._type_col]}\n"
+                f"It should be one of: {', '.join(self.VALID_DATATYPES)}"
+            )
+
+    def _validate_data_file(self, row):
+        """Assert that the datafile is non-empty and has the right format."""
+        if len(row[self._file_col]) <= 0:
+            raise AssertionError("Data file is required.")
+        self._validate_data_format(row[self._file_col])
+
+    def _validate_index_file(self, row):
+        """Assert that the indexfile is non-empty and has the right format."""
+        if len(row[self._index_col]) <= 0:
+            raise AssertionError("Data index file is required.")
+        if row[self._file_col].endswith("bam") and not row[self._index_col].endswith("bai"):
+            raise AssertionError("bai index file should be given for bam file.")
+        if row[self._file_col].endswith("cram") and not row[self._index_col].endswith("crai"):
+            raise AssertionError("crai index file shuld be given for cram file.")
+        self._validate_index_format(row[self._index_col])
+
+    def _validate_data_format(self, filename):
+        """Assert that a given filename has one of the expected read data file extensions."""
+        if not any(filename.endswith(extension) for extension in self.DATA_VALID_FORMATS):
+            raise AssertionError(
+                f"The data file has an unrecognized extension: {filename}\n"
+                f"It should be one of: {', '.join(self.DATA_VALID_FORMATS)}"
+            )
+
+    def _validate_index_format(self, filename):
+        """Assert that a given filename has one of the expected read index file extensions."""
+        if not any(filename.endswith(extension) for extension in self.INDEX_VALID_FORMATS):
+            raise AssertionError(
+                f"The index file has an unrecognized extension: {filename}\n"
+                f"It should be one of: {', '.join(self.INDEX_VALID_FORMATS)}"
             )
 
     def validate_unique_samples(self):
         """
-        Assert that the combination of sample name and FASTQ filename is unique.
+        Assert that the combination of sample name, data and index filename is unique.
 
         In addition to the validation, also rename all samples to have a suffix of _T{n}, where n is the
-        number of times the same sample exist, but with different FASTQ files, e.g., multiple runs per experiment.
+        number of times the same sample exist, but with different files, e.g., multiple runs per experiment.
 
         """
         if len(self._seen) != len(self.modified):
-            raise AssertionError("The pair of sample name and FASTQ must be unique.")
+            raise AssertionError("The combination of sample name, data and index file must be unique.")
```

Comment:
This message would need to be updated if you only use the sample name and the data file, as per my above comment

Reply:
Based on previous comments by @muffato, should this not be changed. From what I understood, you are checking for unique combination of sample name and data file only.

Reply:
I remember I changed here.

Done now.